### DISCUSSION TITLE
Recette v8.4.0

### DISCUSSION THREAD

**Discussion by alexisig** [MTES-MCT/sparte#780] [Recettes]
Note de MAJ liée : https://github.com/MTES-MCT/sparte/releases/tag/v8.4.0

## Infos importantes pour la recette 

- Uniquement les données du département 80 sont disponibles sur l'environnement de `staging` (on peut le modifier si vous voulez tester un autre département).
- Pour chaque élément, il y a parfois plus d'informations / des captures d'écran sur le lien de la PR associée
- L'URL de l'environnement de test : https://sparte-staging.osc-fr1.scalingo.io/
- Les retours sont à faire sur cette discussion, en séparant les retours les uns des autres

### Fonctionnalités
* 🚀 **Ajout d'une nouvelle section de graphiques croisant population et consommation d'espace, disponible pour tous les types de territoire** https://github.com/MTES-MCT/sparte/pull/738
* Ajout des attributions OpenStreetMap sur toutes les cartes statiques et dynamiques https://github.com/MTES-MCT/sparte/pull/747
* Ajout d'une page-formulaire d'inscription à la newsletter https://github.com/MTES-MCT/sparte/pull/755
* Toutes les cartes statiques ont été remplacées par leurs équivalents dynamiques sur l'interface du site https://github.com/MTES-MCT/sparte/pull/750
* La trajectoire de sobriété foncière peut désormais accepter des valeurs décimales https://github.com/MTES-MCT/sparte/pull/736

### Données (partiellement testable)

* Ajout des données OCS GE pour les départements 10, 53, 09 et 44 https://github.com/MTES-MCT/sparte/pull/714, https://github.com/MTES-MCT/sparte/pull/752
* Suppression des diagnostics basés sur des territoires qui n'existent plus dans la dernière version du Code Officiel Géographique (2024) https://github.com/MTES-MCT/sparte/pull/721
* Ajout des données de population et adaptation des données au dernier Code Officiel Géographique (2024) https://github.com/MTES-MCT/sparte/pull/706
* Toutes les opérations utilisant les données de consommation ont été pré-caculées: les performances devraient être grandement améliorées, notamment sur les diagnostics de département ou de région https://github.com/MTES-MCT/sparte/pull/734

### Correctifs
* Correction d'un bug où le total d'artificialisation des zonage d'urbanisme pouvait parfois dépasser 100% https://github.com/MTES-MCT/sparte/pull/713
* Mise à jour des cartes interactives de consommation et d'artificialisation afin de ne montrer que des valeurs relatives en aplat de couleur https://github.com/MTES-MCT/sparte/pull/768
* Correction d'un bug sur le clic qui ne fonctionnait pas sur les résultats de recherche pour une petite partie de nos utilisateurs https://github.com/MTES-MCT/sparte/pull/770
* Renommage de l'onglet trajectoire ZAN par trajectoire de sobriété foncière https://github.com/MTES-MCT/sparte/pull/762
* Suppression de la page tutoriel https://github.com/MTES-MCT/sparte/pull/758
* Correction des chiffres des matrices de passage OCS Ge qui n'étaient parfois pas cohérents avec les tableaux de données associées https://github.com/MTES-MCT/sparte/pull/741 

### Tech (non testabe)
* Remplacement des clefs primaires des territoires par des clefs naturelles https://github.com/MTES-MCT/sparte/pull/652
* Supprime l'étape DBT d'ingest_admin_express https://github.com/MTES-MCT/sparte/pull/710
* Correction des géométries invalides de la table for_app_artificialarea https://github.com/MTES-MCT/sparte/pull/711
* Suppression de l'option subset pour la table for_app_acommunediff https://github.com/MTES-MCT/sparte/pull/712
* Suppression des diagnostics de territoires composites https://github.com/MTES-MCT/sparte/pull/722
* Suppression des objets `Project` sans land_id  https://github.com/MTES-MCT/sparte/pull/723
* Suppression des diagnostics pour lesquels la valeur de `look_a_like` est supérieur à 250 caractères https://github.com/MTES-MCT/sparte/pull/726
* Suppressions des options de la carte de déploiement OCS GE https://github.com/MTES-MCT/sparte/pull/730
* Correction du template de mail du rapport conso : utilisation du nom du diagnostic au lieu du nom du territoire https://github.com/MTES-MCT/sparte/pull/715
* Suppression des anciens filtres de la recherche de territoires https://github.com/MTES-MCT/sparte/pull/735
* Correction des tests dbt de conso / pop https://github.com/MTES-MCT/sparte/pull/739
* Ajout de statistiques de consommation et population différentes pour les communes sans EPCI https://github.com/MTES-MCT/sparte/pull/742
* Correction du tracking matomo https://github.com/MTES-MCT/sparte/pull/733
* Ajout de tests unitaires à airflow https://github.com/MTES-MCT/sparte/pull/753 https://github.com/MTES-MCT/sparte/pull/754
* Ajout des tables UsageSol, CouvertureSol et CouvertureUsageMatrix au dag update_app https://github.com/MTES-MCT/sparte/pull/757
* Retire la possibilité de créer des utilisateurs avec la même adresse email mais avec un casing différent https://github.com/MTES-MCT/sparte/pull/527
* Suppression de la table `CommunePop` https://github.com/MTES-MCT/sparte/pull/761
* Ajout d'un DAG airflow pour monitorer les URLs qui n'existent plus sur la page de téléchargement de l'OCS GE https://github.com/MTES-MCT/sparte/pull/769
* Suppression de la table Cerema https://github.com/MTES-MCT/sparte/pull/768
* Remplacement des sources OCS GE des couches de diff supprimées sur le site de l'IGN (les données ne sont pas encore rechargées, seulement les URLs de source ont été modifiés) https://github.com/MTES-MCT/sparte/pull/774
* Ré-ajoute jupyter au docker-compose https://github.com/MTES-MCT/sparte/pull/772

**Full Changelog**: https://github.com/MTES-MCT/sparte/compare/v8.3.0...v8.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/MTES-MCT/sparte/releases/tag/v8.4.0'>v8.4.0</a>.</em>

**Comment by leabouche** [comment]
Section "Consommation d'espace et démographie": 
Chiffre de consommation: Je remplacerais "D'espaces NAF consommés sur la période" par "Consommation d'espaces NAF de 2011 à 2022" par souci de cohérence avec ce qui est affiché en haut de page.
Chiffre de population: je mettrais "hab" à côté du chiffre, et en-dessous : "Flux de population entre 2011 et 2022" - aussi par souci de cohérence des formats (my bad, j'avais pas mis ca sur la maquette..)

**Comment by leabouche** [comment]
Erreur de chiffre dans le tableau des données sur l'évolution de population (territoire = Amiens)

![Capture d’écran 2024-12-03 à 16 56 23](https://github.com/user-attachments/assets/acae2852-73aa-4987-a7ed-e69e3ed78522)

**Reply by alexisig** [reply]
C'est l'évolution de l'année en cours, qui sera visible sur l'année d'après.  Le flux de cette année est aussi une estimation, mais pas le stock. Peut-être on devrait préciser ?

**Reply by leabouche** [reply]
ah oui ok. à mon avis on ne devrait pas du tout afficher l'estimation de flux, d'autant qu'elle ne figure pas sur le graphe.

**Comment by leabouche** [comment]
J'aime bien l'idée d'avoir la conversion en hab/km2 mais je trouve que l'endroit se prete pas trop à afficher les valeurs (on dirait plus un sous titre du graphe). Peut etre plutot entre parenthèses en-dessous de la valeur en hab/ha au niveau du curseur ? 

![Capture d’écran 2024-12-03 à 17 04 41](https://github.com/user-attachments/assets/5158076c-1048-49b9-a2cb-98bcc61c2fe9)

**Reply by leabouche** [reply]
![Capture d’écran 2024-12-03 à 17 11 16](https://github.com/user-attachments/assets/6d0b6172-dd24-4165-9dbb-71646e37fcfe)

(pas bloquant pour la MEP)

**Reply by smdsgn** [reply]
C'est ce que j'avais fait en premier mais ça fait vraiment trop bizarre d'avoir un chiffre incohérent avec l'échelle du graphique., même entre parenthèses.

**Comment by leabouche** [comment]
Je me pose la question de l'accessibilité des légendes en bleu/violet et rouge, c'est plus clair en terme de compréhension du graphe mais peu lisible je trouve - preneuse de vos avis :) 

![Capture d’écran 2024-12-03 à 17 12 28](https://github.com/user-attachments/assets/7bd0877b-ded6-434c-beb2-5fe4a6fe7016)

**Comment by leabouche** [comment]
Download en PNG des graphes: les titres des fichiers sont tronqués
Par ex: 
- consommation-despace-au.png
- consommation-despace-du.png
- volutions-de-la-consomma.png

**Reply by smdsgn** [reply]
Done

**Comment by leabouche** [comment]
Erreur dans le calcul de la somme des surfaces (x10000, exprimé en m2 au lieu d'ha)

<img width="1105" alt="Capture d’écran 2024-12-12 à 15 45 50" src="https://github.com/user-attachments/assets/f1439208-0451-4f7f-8eac-941e7be97d1c" />

**Reply by smdsgn** [reply]
Bien vu

**Reply by smdsgn** [reply]
Done

**Comment by leabouche** [comment]
Bug sur les communes dont la densité > 250 hab/ha: le curseur n'apparait pas sur le graphe (ex: levallois perret)
<img width="785" alt="Capture d’écran 2024-12-12 à 16 00 12" src="https://github.com/user-attachments/assets/9878285b-b14c-48f5-92f2-bb6724fcf43e" />

**Reply by leabouche** [reply]
<img width="777" alt="Capture d’écran 2024-12-12 à 16 05 13" src="https://github.com/user-attachments/assets/349e57ab-5f6f-4509-8741-1f2482d64f05" />

**Reply by leabouche** [reply]
(à priori que sur ces deux communes)

**Reply by smdsgn** [reply]
Bien vu aussi, je m'étais basé sur Paris pour la densité max avec 21 400 habitants au km². Mais le max c'est Levallois-Perret avec 28 500.

**Reply by smdsgn** [reply]
Done

**Comment by alexisig** [comment]
Je suis sur le bug des données d'artif des zonages d'urbanisme ce matin
